### ISSUE TITLE
Issues with 3D subtitles

### ISSUE THREAD

**Issue by markonjezic** [UniversalMediaServer/UniversalMediaServer#878]
I recently upgraded to the version 6.2.2 from an old version from the 5.x series and noticed that the automatically generated 3D subtitles became almost unusable.

There are two issues, which did not exist before.

- Parts of subtitle lines are not being shown during playback (basically everything before the first comma in the subtitle text). This happens because automatically generated ASS files with 3D subtitles contain syntax error due to a missing comma. In generated 3D ASS file there should be two commas before subtitle text. I don't do Java, so can't create a pull request, but the fix for this is trivial. In convertASSToASS3D() function two lines need to be changed:

```diff
--- SubtitleUtils.java	2016-04-13 23:18:06.000000000 +0200
+++ SubtitleUtils.java.patched	2016-04-13 23:18:06.000000000 +0200
@@ -574,13 +574,13 @@
 							.append("Default,,")
 							.append("0000,")
 							.append(String.format("%04d,", 192 - depth3D))
-							.append("0000,")
+							.append("0000,,")
 							.append(text).append("\n")
 							.append("Dialogue: 0,")
 							.append(timeMatcher.group())
 							.append("Default,,")
 							.append(String.format("%04d,", 192 - depth3D))
-							.append("0000,0000,")
+							.append("0000,0000,,")
 							.append(text).append("\n");
 						}
 					}
```

- 3D subtitles only work properly when you start movie reproduction from the beginning. If you seek forward for example, UMS will subsequently switch to the SRT subtitles that exist in the same directory as the movie, instead of using the generated 3D ASS file. Here's a part from the log file that shows the problem:

```log
DEBUG 2016-04-13 23:49:45.878 [New I/O worker #10] Detected charset "UTF-8" in file C:\ProgramData\UMS\data\subs\TEST.3D.HSBS_ID100_1460567921936.ass
DEBUG 2016-04-13 23:49:45.894 [New I/O worker #10] Subtitles converted to 3DASS format and stored in the file: TEST.3D.HSBS_ID100_1460567921936_3D.ass
DEBUG 2016-04-13 23:49:45.894 [New I/O worker #10] Creating pipe \\.\pipe\ffmpegvideo_61_1460584185894
DEBUG 2016-04-13 23:49:45.894 [Thread-31] Waiting for pipe connection \\.\pipe\ffmpegvideo_61_1460584185894
DEBUG 2016-04-13 23:49:46.206 [ffmpeg.exe-3] Starting "C:\Program Files (x86)\Universal Media Server\win32\ffmpeg.exe" -y -loglevel fatal -ss 2217.23 -i F:\TEST\TEST.3D.HSBS.mp4 -filter_complex setpts=PTS+2217.23/TB,subtitles=C\\:/ProgramData/UMS/data/subs/TEST.3D.HSBS_ID100_1460567921936_3D.ass,setpts=PTS-STARTPTS -t 2218.23 -g 25 -qmax 7 -qmin 2 -ab 448k -ar 48000 -c:a ac3 -c:v mpeg2video -f vob \\.\pipe\ffmpegvideo_61_1460584185894
DEBUG 2016-04-13 23:49:46.206 [ffmpeg.exe-3] Reading pipe: \\.\pipe\ffmpegvideo_61_1460584185894
DEBUG 2016-04-13 23:49:46.237 [ffmpeg.exe-3] Attaching thread: ffmpeg.exe-3
DEBUG 2016-04-13 23:49:46.362 [Thread-31] Result of \\.\pipe\ffmpegvideo_61_1460584185894 : true
DEBUG 2016-04-13 23:49:48.140 [New I/O worker #11] Matched media renderer "BRAVIA KDL-55W815B" based on address /10.10.10.127
DEBUG 2016-04-13 23:49:48.140 [New I/O worker #11] Requesting time seek: 2217.23 seconds
DEBUG 2016-04-13 23:49:48.140 [External Process Stopper] Stopping process: ffmpeg.exe-3
DEBUG 2016-04-13 23:49:48.140 [New I/O worker #11] Creating pipe \\.\pipe\ffmpegvideo_62_1460584188140
DEBUG 2016-04-13 23:49:48.140 [Thread-37] Waiting for pipe connection \\.\pipe\ffmpegvideo_62_1460584188140
DEBUG 2016-04-13 23:49:48.452 [ffmpeg.exe-4] Starting "C:\Program Files (x86)\Universal Media Server\win32\ffmpeg.exe" -y -loglevel fatal -ss 2217.23 -i F:\TEST\TEST.3D.HSBS.mp4 -filter_complex setpts=PTS+2217.23/TB,subtitles=F\\:/TEST/TEST.3D.HSBS.srt,setpts=PTS-STARTPTS -bufsize 7000000 -maxrate 29000000 -g 25 -qmax 7 -qmin 2 -ab 448k -ar 48000 -c:a ac3 -c:v mpeg2video -f vob \\.\pipe\ffmpegvideo_62_1460584188140
DEBUG 2016-04-13 23:49:48.452 [ffmpeg.exe-4] Reading pipe: \\.\pipe\ffmpegvideo_62_1460584188140
DEBUG 2016-04-13 23:49:48.452 [ffmpeg.exe-4] Attaching thread: ffmpeg.exe-4
DEBUG 2016-04-13 23:49:48.608 [Thread-37] Result of \\.\pipe\ffmpegvideo_62_1460584188140 : true
DEBUG 2016-04-13 23:49:50.230 [cling-52] TransportState: PLAYING
DEBUG 2016-04-13 23:49:50.418 [New I/O worker #12] Matched media renderer "BRAVIA KDL-55W815B" based on address /10.10.10.127
```

You can clearly see that the first call to the ffpmeg.exe [ffmpeg.exe-3] uses the generated 3D ASS file, while after seeking forward, next call to the ffpmeg.exe [ffmpeg.exe-4] has switched to the original SRT file and as a result subtitles are no longer readable on screen. I took a brief look at the source code, but since I'm not that familiar with UMS I can't pinpoint where the problem lies.

**Comment by valib** [comment]
@markonjezic thanks for the input. Hopefully fixed in 4981ec1, 43c3ec4  and 2905f82

**Comment by SubJunk** [comment]
I assume this is closed now, feel free to reopen if it isn't fixed
